src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -191,13 +191,29 @@ public RuntimeConfig(
         this.Entities = Entities;
         this.DefaultDataSourceName = Guid.NewGuid().ToString();
 
+        if (this.DataSource is null)
+        {
+            throw new DataApiBuilderException(
+                message: "DataSource is a mandatory property in DAB Config",
+                statusCode: HttpStatusCode.UnprocessableEntity,
+                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+        }
+
         // we will set them up with default values
         _dataSourceNameToDataSource = new Dictionary<string, DataSource>
         {
             { this.DefaultDataSourceName, this.DataSource }
         };
 
         _entityNameToDataSourceName = new Dictionary<string, string>();
+        if (Entities is null)
+        {
+            throw new DataApiBuilderException(
+                message: "Entities is a mandatory property in DAB Config",
```

Comment:
Message also needs to be a bit more specific to show entities needs to be in all config files
```suggestion
                message: "Entities is a mandatory property in all DAB configuration files",
```

Reply:
Since we are validating config file passed in the parameter c {configfilename} which is a single file not multiple. This shouldnt be needed

Reply:
Dont we traverse through all the config files?

Reply:
No, we only validate top level data source file. This is the enhancement we need to take up later to validate all config files in the multiple data source scenario